qemu/tests/balloon_check.py

```diff
@@ -77,8 +77,18 @@ def balloon_memory(self, new_mem):
         :type new_mem: int
         """
         error.context("Change VM memory to %s" % new_mem, logging.info)
-        compare_mem = new_mem
-        self.vm.balloon(new_mem)
+        min_sz, max_sz = self.get_memory_boundary()
+        if new_mem > self.ori_mem:
+            compare_mem = self.ori_mem
+        elif new_mem == 0:
+            compare_mem = self.current_mmem
+        else:
+            compare_mem = new_mem
+        try:
+            self.vm.balloon(new_mem)
+        except Exception, e:
+            if not self.params.get('illegal_value_check') and new_mem != self.current_mmem:
+                logging.error("balloon memory fail with error message: %s" %e)
```

Comment:
better to write as:
msg = "balloon memory fail with error message: %s" %e
raise error.TestFail(msg)